crates/bevy_asset/src/handle.rs

```diff
@@ -198,7 +198,7 @@ impl<A: Asset> Handle<A> {
     }
 }
 
-impl<A: Asset> Default for Handle<A> {
+impl<A> Default for Handle<A> {
```

Comment:
Why should removing `A: Asset` from `Handle` / `AssetId` require allowing for default construction without that type parameter? Surely this now allows for the creation of impossible handles?

```rust
let impossible_handle: Handle<NotAnAsset> = default();
```

Reply:
Yea this is something I would like to add back, though I think it would require either manually implementing `Reflect` or adding some way for the derive macro to add bounds on the generic type.

Reply:
@bushrat011899 What do you mean by "impossible handles"? Even if you construct a `Handle<NotAnAsset>`, it wouldn't be usable with `Assets<NotAnAsset>` due to trait bounds.

Reply:
> @bushrat011899 What do you mean by "impossible handles"? Even if you construct a `Handle<NotAnAsset>`, it wouldn't be usable with `Assets<NotAnAsset>` due to trait bounds.

As in it is currently impossible to describe a type `Handle<NotAnAsset>`, and impossible to create a value `Handle<NotAnAsset>` in Bevy. This change doesn't just permit the type `Handle<NotAnAsset>`, it permits a runtime value of `Handle::<NotAnAsset>::default()`.

Yes it cannot be used in `Assets`, and it won't point to a valid asset, so it's mostly a dead-end value, just like `Handle::<ActuallyAnAsset>::default()` is. It's just something that shouldn't exist.